src/NServiceBus.Transport.Msmq/MsmqConfigurationExtensions.cs

```diff
@@ -241,6 +241,19 @@ public static TransportExtensions<MsmqTransport> DisableNativeTimeToBeReceivedIn
             return transport;
         }
 
+        /// <summary>
+        /// Configures native delayed delivery.
+        /// </summary>
+        [PreObsolete(
+            ReplacementTypeOrMember = "Transport.DelayedDelivery",
+            RemoveInVersion = "4",
+            TreatAsErrorFromVersion = "3")]
+        public static DelayedDeliverySettings NativeDelayedDelivery(this TransportExtensions<MsmqTransport> config, IDelayedMessageStore delayedMessageStore)
+        {
+            config.Transport.DelayedDelivery = new DelayedDeliverySettings(delayedMessageStore);
```

Comment:
Should there be some kind of guard here to prevent passing a null `delayedMessageStore`?

Reply:
Added a Guard to check for that.